src/integrationTest/samVisualize/visualizeSamTemplate.test.ts

```diff
@@ -0,0 +1,152 @@
+                    Properties:
+                        Path: /hello
+                        Method: get`
+/**
+ * Helper function to create a text file with supplied content and open it with a TextEditor
+ *
+ * @param fileText The supplied text to fill this file with
+ * @param fileName The name of the file to save it as. Include the file extension here.
+ *
+ * @returns TextEditor that was just opened
+ */
+async function openATextEditorWithText(fileText: string, fileName: string): Promise<vscode.TextEditor> {
+    const completeFilePath = path.join(tempFolder, fileName)
+    await fs.writeFile(completeFilePath, fileText)
+
+    const textDocument = await vscode.workspace.openTextDocument(completeFilePath)
+
+    return await vscode.window.showTextDocument(textDocument)
+}
+let tempFolder: string
+describe('visualizeSamTemplate', async function () {
+    beforeEach(async function () {
+        // Make a temp folder for all these tests
+        tempFolder = await makeTemporaryToolkitFolder()
+    })
+
+    afterEach(async function () {
+        await fs.remove(tempFolder)
+    })
+
+    after(async function () {
+        // Test suite cleans up after itself
+        await closeAllEditors()
+    })
+
+    it('correctly displays content when given an active editor containing a valid SAM Template', async function () {
+        await openATextEditorWithText(sampleSamTemplateYaml, sampleSamTemplateFileName)
+        const result = await vscode.commands.executeCommand<vscode.WebviewPanel>('aws.samVisualize.renderTemplate')
+
+        assert.ok(result)
+        assert.deepStrictEqual(result.title, `${sampleSamTemplateFileName} (Rendering)`)
+        assert.deepStrictEqual(result.viewType, 'samVisualization')
+        assert.ok(result.webview.html)
+    })
+
+    it('update webview when the editor is changed', async function () {
+        const textEditor = await openATextEditorWithText('Sample Text', sampleSamTemplateFileName)
+
+        const result = await vscode.commands.executeCommand<vscode.WebviewPanel>('aws.samVisualize.renderTemplate')
+
+        assert.ok(result)
+
+        const didViewStateChange = new Promise<vscode.WebviewPanelOnDidChangeViewStateEvent>(resolve => {
+            result.onDidChangeViewState(event => {
+                assert.ok(event)
+                resolve(event)
+            })
+        })
+
+        // Edit the editor to replace the entire file with a working SAM template
+        await textEditor.edit(edit => {
+            edit.replace(
+                new vscode.Range(
+                    textEditor.document.positionAt(0),
+                    textEditor.document.positionAt(textEditor.document.getText().length)
+                ),
+                sampleSamTemplateYaml
+            )
+        })
+
+        await didViewStateChange
+    })
+
+    it('throws an error if no text editor is open', async function () {
+        // Ensure no text editor is open
+        await closeAllEditors()
```

Comment:
Is there a reason to not have this in the `afterEach`?

Reply:
It's not necessary to close all the editors after each test, but it can go in the `afterEach` and I suppose it'll prevent too many open editors if more tests are added.